webui-src/app/forums/forums_util.js

```diff
@@ -0,0 +1,219 @@
+      },
+      (data) => {
+        details = data.forumsInfo[0];
+      }
+    )
+    .then(() => {
+      if (Data.DisplayForums[keyid] === undefined) {
+        Data.DisplayForums[keyid] = {
+          name: details.mMeta.mGroupName,
+          isSearched: true,
+          description: details.mDescription,
+          isSubscribed: details.mMeta.mSubscribeFlags === GROUP_SUBSCRIBE_SUBSCRIBED,
+        };
+      }
+    });
+//   console.log(Data.DisplayForums[keyid]);
+}
+const DisplayForumsFromList = () => {
+  return {
+    oninit: (v) => {},
+    view: (v) =>
+      m(
+        'tr',
+        {
+          key: v.attrs.id,
+          class:
+            Data.DisplayForums[v.attrs.id] && Data.DisplayForums[v.attrs.id].isSearched
+              ? ''
+              : 'hidden',
+          onclick: () => {
+            m.route.set('/forums/:tab/:mGroupId', {
+              tab: v.attrs.category,
+              mGroupId: v.attrs.id,
+            });
+          },
+        },
+        [m('td', Data.DisplayForums[v.attrs.id] ? Data.DisplayForums[v.attrs.id].name : '')]
+      ),
+  };
+};
+
+const ForumSummary = () => {
+  let keyid = {};
+  return {
+    oninit: (v) => {
+      keyid = v.attrs.details.mGroupId;
+      updateDisplayForums(keyid);
+    },
+
+    view: (v) => {},
+  };
+};
+
+const MessageView = () => {
+  let fname = '';
+  let fsubscribed = {};
+  let toggleUnsubscribe = false;
+  return {
+    oninit: (v) => {
+      if (Data.DisplayForums[v.attrs.id]) {
+        fname = Data.DisplayForums[v.attrs.id].name;
+        fsubscribed = Data.DisplayForums[v.attrs.id].isSubscribed;
+      }
+    },
+    view: (v) =>
+      m(
+        '.widget',
+        {
+          key: v.attrs.id,
+        },
+        [
+          m(
+            'a[title=Back]',
+            {
+              onclick: () =>
+                m.route.set('/forums/:tab', {
+                  tab: m.route.param().tab,
+                }),
+            },
+            m('i.fas.fa-arrow-left')
+          ),
+          m(
+            'button',
+            {
+              onclick: () => {
+                if (!fsubscribed) {
+                  rs.rsJsonApiRequest(
+                    '/rsgxsforums/subscribeToForum',
+                    {
+                      forumId: v.attrs.id,
+                      subscribe: true,
+                    },
+                    (data) => {
+                      console.log(data);
+                    }
+                  ).then(() => {
+                    Data.DisplayForums[v.attrs.id].isSubscribed = true;
+                    fsubscribed = true;
+                  });
+                } else {
+                  toggleUnsubscribe = !toggleUnsubscribe;
+                }
+              },
+            },
+            fsubscribed ? 'Subscribed' : 'Subscribe'
+          ),
+          m(
+            'button[id=toggleunsub]',
+            {
+              style: 'display:' + (toggleUnsubscribe ? 'block' : 'none'),
```

Comment:
You actually don't need to do string patching, style can also be an object.
See https://mithril.js.org/hyperscript.html#style-attribute
```suggestion
              style: { display: toggleUnsubscribe ? 'block' : 'none' },
```